include/zio/zio_dev.h

```diff
@@ -141,6 +147,37 @@ struct zio_dev_chan {
 
 };
 
+/**
+ * @brief Pre-defined attribute types with the ability to extend by a driver
+ * implementation.
+ *
+ * Extended values should always be based on ZIO_ATTR_TYPES count with an
+ * additive and defined in a publicly available driver header.
+ *
+ * Example: #define MY_ATTR_TYPE (ZIO_ATTR_TYPES+1)
+ */
+enum zio_attr_type {
+	/* samples per second */
+	ZIO_SAMPLE_RATE,
+
+	/* A count of attribute types always begins the user definable range */
+	ZIO_ATTR_TYPES,
+
+	/* enforce a specific enum size allowing for a good number of user
+	 * definable values but keeping
+	 */
+	ZIO_ATTR_TYPE_MAKE_16_BIT = 0xFFFF
+};
+
+/** Attribute record. */
+struct zio_attr_desc {
+	/* Attribute type, describing what it is such as the sample rate */
+	enum zio_attr_type type;
```

Comment:
Doxygen comment blocks need to start with `**`, and would benefit from the `@brief` tag before the text.

Struct attributes generally have the comment **after** the field, though, with `/**<` to indicate that the comment block refers to the preceding field, so:

```
enum zio_attr_type type;
/**< @brief Attribute type, describing what it is such as the sample rate */

...
```